ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/block/BlockProcessorElectra.java

```diff
@@ -126,30 +135,41 @@ protected void verifyOutstandingDepositsAreProcessed(
   }
 
   @Override
-  protected void processExecutionLayerExits(
+  protected void processExecutionLayerWithdrawRequests(
       final MutableBeaconState state,
       final Optional<ExecutionPayload> executionPayload,
       final Supplier<ValidatorExitContext> validatorExitContextSupplier)
       throws BlockProcessingException {
-    processExecutionLayerExits(
-        state, getExecutionLayerExitsFromBlock(executionPayload), validatorExitContextSupplier);
+    this.processExecutionLayerWithdrawRequests(
+        state,
+        getExecutionLayerWithdrawRequestsFromBlock(executionPayload),
+        validatorExitContextSupplier);
   }
 
-  /*
-    Implements process_execution_layer_exit from consensus-specs (EIP-7002)
-  */
+  /**
+   * Implements process_execution_layer_withdraw_request from consensus-specs (EIP-7002 & EIP-7251).
+   */
   @Override
-  public void processExecutionLayerExits(
+  public void processExecutionLayerWithdrawRequests(
       final MutableBeaconState state,
-      final SszList<ExecutionLayerExit> exits,
+      final SszList<ExecutionLayerWithdrawRequest> withdrawRequests,
       final Supplier<ValidatorExitContext> validatorExitContextSupplier)
       throws BlockProcessingException {
     final UInt64 currentEpoch = miscHelpers.computeEpochAtSlot(state.getSlot());
 
-    exits.forEach(
-        exit -> {
+    withdrawRequests.forEach(
+        withdrawRequest -> {
+          // If partial withdrawal queue is full, only full exits are processed
+          final boolean isFullExitRequest = withdrawRequest.getAmount().isZero();
```

Comment:
I'm puzzled by this check. Seems different from spec:
```
is_full_exit_request = amount == FULL_EXIT_REQUEST_AMOUNT
```

Am I missing something?

Reply:
You might be missing that [`FULL_EXIT_REQUEST_AMOUNT` is zero](https://github.com/ethereum/consensus-specs/blob/376683d935dcaf3604636a1bd382a1412665b0a8/specs/electra/beacon-chain.md?plain=1#L111). We should definitely add a constant for that somewhere, rather than the `isZero()` call. If not, what exactly is puzzling you?

Reply:
> You might be missing that FULL_EXIT_REQUEST_AMOUNT is zero

That is what I was missing :)

Reply:
Added a constant for that, should be more clear now 😄